services:
  fe_api_db:
    image: mysql:8.0
    container_name: e-love-fe-api-db
    env_file:
      - ./docker/db-app/db.env
    volumes:
      - fe_api_db_data:/var/lib/mysql
    ports:
      - "3306:3306"
    networks:
      - e-love-network
    healthcheck:
      test: "mysqladmin ping -h localhost"
      interval: 10s
      timeout: 5s
      retries: 5

  e-love-fe-api:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        INSTALL_DEV: "true"
    container_name: fe-api
    ports:
      - "8080:8000"
    env_file:
      - .env
      - ./docker/chat.env
    volumes:
      - type: bind
        source: ./api
        target: /app/api
      - type: bind
        source: ./auth
        target: /app/auth
      - type: bind
        source: ./configuration
        target: /app/configuration
      - type: bind
        source: ./core
        target: /app/core
      - type: bind
        source: ./alembic
        target: /app/alembic
      - type: bind
        source: ./alembic.ini
        target: /app/alembic.ini
      - type: bind
        source: ./docker
        target: /app/docker
      - type: bind
        source: ./main.py
        target: /app/main.py
    networks:
      - e-love-network
    depends_on:
      fe_api_db:
        condition: service_healthy

networks:
  e-love-network:
    external: true

volumes:
  fe_api_db_data:
    driver: local
